webui/react/src/types.ts

```diff
@@ -367,6 +367,34 @@ export type HyperparameterType = t.TypeOf<typeof HyperparametersType>;
 export type Hyperparameters = {
   [keys: string]: Hyperparameters | HyperparameterBase;
 };
+
+export const IntegrationBase = t.type({
+  dataset: t.type({
+    branch: t.string,
+    commit: t.string,
+    project: t.string,
+    repo: t.string,
+    token: t.string,
+  }),
+  proxy: t.type({
+    host: t.string,
+    port: t.number,
+    scheme: t.string,
+  }),
+});
```

Comment:
this is fine, but i'm not sure other integrations will have the same setup -- we could just inline this instead?